//sgl/devel/grimoire/x11-toolkits/wxGTK/gtk-2.4-fix.patch#1 - add change 30747 (text)
diff -Nur wxGTK-2.4.2/src/gtk/menu.cpp wxGTK-2.4.2-fix/src/gtk/menu.cpp
--- wxGTK-2.4.2/src/gtk/menu.cpp	2003-09-21 13:31:57.000000000 +0200
+++ wxGTK-2.4.2-fix/src/gtk/menu.cpp	2004-03-29 14:53:34.000000000 +0200
@@ -30,13 +30,13 @@
 #ifdef __WXGTK20__
     #include <glib-object.h>
 
-    #define gtk_accel_group_attach(g, o) _gtk_accel_group_attach((g), (o))
-    #define gtk_accel_group_detach(g, o) _gtk_accel_group_detach((g), (o))
+    #define gtk_accel_group_attach(g, o) gtk_window_add_accel_group((o), (g))
+    #define gtk_accel_group_detach(g, o) gtk_window_remove_accel_group((o), (g))
     #define gtk_menu_ensure_uline_accel_group(m) gtk_menu_get_accel_group(m)
 
-    #define ACCEL_OBJECT        GObject
+    #define ACCEL_OBJECT        GtkWindow
     #define ACCEL_OBJECTS(a)    (a)->acceleratables
-    #define ACCEL_OBJ_CAST(obj) G_OBJECT(obj)
+    #define ACCEL_OBJ_CAST(obj) GTK_WINDOW(obj)
 #else // GTK+ 1.x
     #define ACCEL_OBJECT        GtkObject
     #define ACCEL_OBJECTS(a)    (a)->attach_objects
diff -Nur wxGTK-2.4.2/src/gtk/settings.cpp wxGTK-2.4.2-fix/src/gtk/settings.cpp
--- wxGTK-2.4.2/src/gtk/settings.cpp	2003-09-21 13:31:57.000000000 +0200
+++ wxGTK-2.4.2-fix/src/gtk/settings.cpp	2004-03-29 15:15:35.000000000 +0200
@@ -340,8 +340,8 @@
                 }  
                 else  
                 {  
-                    const gchar *font_name =
-                        _gtk_rc_context_get_default_font_name(gtk_settings_get_default());
+                    gchar *font_name;
+                    g_object_get(gtk_settings_get_default(), "gtk-font-name", &font_name, NULL);
                     g_systemFont = new wxFont(wxString::FromAscii(font_name));
                 }  
                 gtk_widget_destroy( widget );
